code/modules/atmospherics/machinery/trinary/manifold_valve.dm

```diff
@@ -0,0 +1,140 @@
+
+/obj/machinery/atmospherics/trinary/manifold_valve
+	icon = 'icons/obj/atmospherics/manifold_valve.dmi'
+	icon_state = "manifold_valve0"
+
+	name = "manifold valve"
+	desc = "A pipe valve"
+
+	var/divert = FALSE
+
+	var/frequency = FREQ_AIR_ALARM_CONTROL
+	var/id = null
+
+/obj/machinery/atmospherics/trinary/manifold_valve/New()
+	..()
+	qdel(src.air1)
+	src.air1 = null
+	qdel(src.air2)
+	src.air2 = null
+	qdel(src.air3)
+	src.air3 = null
```

Comment:
not sure why this nullifying is necessary?

Reply:
we don't use these gasmixes, so i disposed of them to save memory. It's explicitly set to null to clear our reference to it.

Reply:
fair enough